<a href="https://colab.research.google.com/github/BenjaminEngel919/Medical-Data/blob/main/Project_Two_Part_One.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Benjamin Engel

In [2]:
# Pandas
import pandas as pd
# Numpy
import numpy as np
# MatplotLib
import matplotlib.pyplot as plt

# Preprocessing
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.compose import make_column_selector
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.pipeline import make_pipeline
from sklearn.compose import make_column_transformer

# Models
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier

# Classification Metrics
from sklearn.metrics import accuracy_score, recall_score, precision_score, \
f1_score, classification_report, confusion_matrix, ConfusionMatrixDisplay


# Set global scikit-learn configuration 
from sklearn import set_config
# Display estimators as a diagram
set_config(display='diagram') # 'text' or 'diagram'}

In [3]:
## Import and mount drive from Google Colab
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#First Choice: Dataset 1
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?


In [4]:
# Load the data
stroke = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/healthcare-dataset-stroke-data.csv')

In [5]:
stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

##1. Source of Data is a Kaggle Data Set

##2. According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.



##3. Target is 'stroke'

##4. One row represents One person.

##5. This is a Classification problem

##6. How many features does the data have?

In [11]:
stroke_types = stroke.dtypes
str_cols = stroke_types[ stroke_types=='object'].index
str_cols

Index(['gender', 'ever_married', 'work_type', 'Residence_type',
       'smoking_status'],
      dtype='object')

###Five total features.

##7. How many rows are in the data set.

In [6]:
## Display the number of rows and columns for the dataframe
stroke.shape
print(f'There are {stroke.shape[0]} rows, and {stroke.shape[1]} columns.')

There are 5110 rows, and 12 columns.


# 5110 Rows

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

###The amount of data is sufficient. Challenges I forsee is choosing the proper features for the target. Lots of two choice columns without a lot of variance.

#Second Choice: Dataset 2
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?


In [8]:
# Load the data
cirrhosis = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/cirrhosis.csv')

In [9]:
cirrhosis.head()

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

##1. Source of Data is a Kaggle Data Set

##2. Cirrhosis is a late stage of scarring (fibrosis) of the liver caused by many forms of liver diseases and conditions, such as hepatitis and chronic alcoholism. 
The following data contains the information collected from the Mayo Clinic trial in primary biliary cirrhosis (PBC) of the liver conducted between 1974 and 1984.


##3. Target is 'Status'

##4. One row represents One person.

##5. This is a Classification problem

##6. How many features does the data have?

In [12]:
cirrhosis_types = cirrhosis.dtypes
str_cols = cirrhosis_types[ cirrhosis_types=='object'].index
str_cols

Index(['Status', 'Drug', 'Sex', 'Ascites', 'Hepatomegaly', 'Spiders', 'Edema'], dtype='object')

###Seven total features.

##7. How many rows are in the data set.

In [10]:
## Display the number of rows and columns for the dataframe
cirrhosis.shape
print(f'There are {cirrhosis.shape[0]} rows, and {cirrhosis.shape[1]} columns.')

There are 418 rows, and 20 columns.


#There are 418 Rows.



##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

###This data set has a low amount of total rows. Choosing the correct features. Lots of problems with missing data and cleaning to make the data match within the data to model this dataset.